packages/components/src/dropdown-menu/index.js

```diff
@@ -146,50 +145,54 @@ function DropdownMenu( dropdownMenuProps ) {
 				return (
 					<NavigableMenu { ...mergedMenuProps } role="menu">
 						{ isFunction( children ) ? children( props ) : null }
-						{ flatMap( controlSets, ( controlSet, indexOfSet ) =>
-							controlSet.map( ( control, indexOfControl ) => (
-								<Button
-									key={ [
-										indexOfSet,
-										indexOfControl,
-									].join() }
-									onClick={ ( event ) => {
-										event.stopPropagation();
-										props.onClose();
-										if ( control.onClick ) {
-											control.onClick();
+						{ controlSets
+							?.map( ( controlSet, indexOfSet ) =>
```

Comment:
Could we just do [`flatMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) on these instead of chaining `map` and `flat`?

```suggestion
							?.flatMap( ( controlSet, indexOfSet ) =>
```

Reply:
Sure, done in 82d0d88f5a506160f2c0635b07fd84e8f5f2dc9e.

Reply:
Also updated for #42223 in https://github.com/WordPress/gutenberg/pull/42223/commits/072a267a7ef38cea8787067789c269ada074146c.